packages/rum-core/src/domain/assembly.ts

```diff
@@ -68,6 +65,33 @@ export function startRumAssembly(
   buildCommonContext: () => CommonContext,
   reportError: (error: RawError) => void
 ) {
+  MODIFIABLE_FIELD_PATHS_BY_EVENT = {
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏use camelcase for this variable

<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏do we really need to detail each field for each event type? Isn't what we did before enough? I feel like this is extra verbose for not much benefit.

Reply:
> Isn't what we did before enough?

I started with it but with the new possibility of adding attributes that are not on the original object, I was concerned that we unexpectedly allow to add a modifiable attribute on the wrong event type.
I tried to express that with this test:  https://github.com/DataDog/browser-sdk/pull/2186/files#diff-0cc30fdbf6d153bf8bacf2743993dbfbe6cfc8441d98ec7cf714a6213614d393R246

wdyt?

Reply:
Ah right I understand that there is a higher change of misusing those fields now. Sounds good to me then!